google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -16,40 +16,94 @@
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessOutputLineListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.process.ProcessStartListener;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.sdk.CloudSdk;
+import com.google.cloud.tools.intellij.CloudToolsPluginInfoService;
+import com.google.cloud.tools.intellij.login.CredentialedUser;
+import com.google.cloud.tools.intellij.login.Services;
 import com.google.cloud.tools.intellij.stats.UsageTrackerProvider;
+import com.google.cloud.tools.intellij.util.GctBundle;
 import com.google.cloud.tools.intellij.util.GctTracking;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import com.google.gdt.eclipse.login.common.GoogleLoginState;
+import com.google.gson.Gson;
 
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.diagnostic.Logger;
 import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vcs.impl.CancellableRunnable;
 import com.intellij.remoteServer.runtime.Deployment;
 import com.intellij.remoteServer.runtime.deployment.DeploymentRuntime;
-import com.intellij.remoteServer.runtime.deployment.DeploymentRuntime.UndeploymentTaskCallback;
 import com.intellij.remoteServer.runtime.deployment.ServerRuntimeInstance.DeploymentOperationCallback;
 import com.intellij.remoteServer.runtime.log.LoggingHandler;
 
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.File;
+import java.io.IOException;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.Map;
 
 /**
  * A Cloud SDK (gcloud) based implementation of the {@link AppEngineHelper} interface.
  */
 public class CloudSdkAppEngineHelper implements AppEngineHelper {
 
+  private static final Logger logger = Logger.getInstance(CloudSdkAppEngineHelper.class);
+
   private static final String DEFAULT_APP_YAML_PATH = "/generation/src/appengine/mvm/app.yaml";
   private static final String DEFAULT_JAR_DOCKERFILE_PATH
       = "/generation/src/appengine/mvm/jar.dockerfile";
   private static final String DEFAULT_WAR_DOCKERFILE_PATH
       = "/generation/src/appengine/mvm/war.dockerfile";
 
+  private final Project project;
   private final File gcloudCommandPath;
+  private final Environment environment;
+
+  private File credentialsPath;
+
+  public enum Environment {
+    APP_ENGINE_STANDARD("App Engine standard"),
+    APP_ENGINE_FLEX("App Engine flexible");
+
+    private String label;
+
+    Environment(String label) {
+      this.label = label;
+    }
+
+    @Override
+    public String toString() {
+      return label; // TODO message bundle
+    }
+  }
 
-  public CloudSdkAppEngineHelper(@NotNull File gcloudCommandPath) {
+  /**
+   * Initialize the helper.
```

Comment:
Definitely leave out method and constructor javadocs if they don't feel like they add any information.